docs/specifications/tests/Zone-TP/zone01.md

```diff
@@ -3,43 +3,252 @@
+   5. From the DNS response, add the SOA SERIAL field and name server IP address to
+      the *SERIAL Nameservers* set.
+   6. Go to next name server IP address.
+
+5. If the set *MNAME Is Localhost* is non-empty, then output
+   *[Z01_MNAME_IS_LOCALHOST]* with name server(s) IP address(es).
+
+6. If the set *MNAME Is Dot* is non-empty, then output
+   *[Z01_MNAME_IS_DOT]* with name server(s) IP address(es).
+
+7. If the set *MNAME Nameservers* is empty, then terminate these procedures.
+
+8. Else, for each SOA MNAME name server name in *MNAME Nameservers* do:
+   1. Do a [DNS Lookup] of SOA MNAME name server name (A and AAAA) and add the
+      SOA MNAME name server IP address(es) found to the *MNAME Nameservers* set
+      for the same entry as the SOA MNAME name server name.
+   2. If at least one IP address from the previous step was found, then:
+      1. For each SOA MNAME name server IP for the SOA MNAME name server name do: 
+         1. Send *SOA Query* to the name server and collect the response.
+            1. If there is a DNS response with [RCODE Name] "NoError", then:
+               1. If the AA flag is not set, then output *[Z01_MNAME_NOT_AUTHORITATIVE]* 
+                  with SOA MNAME name server name and IP address, and name server(s) IP
+                  address(es) from the *SERIAL Nameservers* set.
+               2. Else, add the SOA SERIAL field to the *MNAME Nameservers* set
+                  for the same entry as the SOA MNAME name server name.
+            2. Else if there is a DNS response with [RCODE Name] other than "NoError", then
+               output *[Z01_MNAME_UNEXPECTED_RCODE]* with [RCODE Name], SOA name server name and
+               IP address, and name server(s) IP address(es) from the *SERIAL Nameservers* set.
+            3. Else, output *[Z01_MNAME_NO_RESPONSE]* with SOA MNAME name server name and IP
+               address, and name server(s) IP address(es) from the *SERIAL Nameservers* set.
+         2. Go to next SOA MNAME name server IP.
+   3. Else, output *[Z01_MNAME_NOT_RESOLVE]* with SOA MNAME name server name,
+      and name server(s) IP address(es) from the *SERIAL Nameservers* set.
+   4. Go to next SOA MNAME name server name.
+
+9. If there is no SOA SERIAL in the *MNAME Nameservers* set, then
+   terminate these procedures.
+
+10. For each SOA SERIAL in *MNAME Nameservers* do:
+    1. For each SOA SERIAL in *SERIAL Nameservers* do:
+       1. Compare both SOA SERIAL values (using the arithmetic in [RFC1982]).
+          1. If the one from *SERIAL Nameservers* is higher, then add SOA MNAME name
+             server name and IP address to the *MNAME Not Master* set, and go to next
+             SOA SERIAL (from the *MNAME Nameservers* set).
+          2. Go to next SOA SERIAL.
+    2. If the set *MNAME Not Master* is empty, then add SOA MNAME name server name and
+       IP address to the *MNAME Is Master* set and exit this loop.
```

Comment:
If there are two different SOA MNAME, then one could meet your criteria, but the other not. I do not think your steps handle that.

Reply:
Right, I used to think there could only be one. Updated.